erts/emulator/sys/unix/erl_child_setup.c

```diff
@@ -629,53 +635,33 @@ main(int argc, char *argv[])
         if (FD_ISSET(sigchld_pipe[0], &read_fds)) {
             int ibuff[2];
             ErtsSysForkerProto proto;
+            ErtsSysExitStatus est, *es;
             res = read_all(sigchld_pipe[0], (char *)ibuff, sizeof(ibuff));
             if (res <= 0) {
                 ABORT("Failed to read from sigchld pipe: %d (%d)", res, errno);
             }
 
-            proto.u.sigchld.port_id = get_port_id((pid_t)(ibuff[0]));
-
-            if (proto.u.sigchld.port_id == THE_NON_VALUE)
```

Comment:
`port_id` is always present, now.  `es` can still be missing from the hash, but only in edge cases like double-close or other bad situations.

Might even be worthwhile to emit a debug line if the child isn't found in forker_hash.